### PR TITLE
feat: add error reporting to Sentry [ROAD-471]

### PR DESCRIPTION
**Pull request by michelkaporin**
- Adds ErrorReporter to capture errors in Sentry
   - Both caught and uncaught exceptions. Unhandled rejections won't be captured.
- Adds ErrorHandler to wrap error handling logic by logging to the output channel and reporting error to Sentry
- Extracts ReportModule to SnykCodeErrorHandler to keep the retry logic for `code-client` and separate from the general error handling
- Adds `snyk.config.local.json` to provide API keys for local extension development.

### CODE REVIEW

File: src/snyk/base/modules/interfaces.ts [hunk 1]
```diff
@@ -17,25 +16,19 @@ export interface IBaseSnykModule {
...
   checkAdvancedMode(): Promise<void>;
 }
 
-export interface IExtension extends IBaseSnykModule, IReportModule, ISnykLib {
+export interface IExtension extends IBaseSnykModule, ISnykLib {
   context: ExtensionContext | undefined;
   activate(context: vscode.ExtensionContext): void;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (bastiandoetsch) [inline src/snyk/base/modules/interfaces.ts:40]: Maybe reactivate that comment?

File: src/snyk/common/error/errorReporter.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import { IVSCodeEnv } from '../vscode/env';
+import { OnUncaughtException } from './integrations/onUncaughtException';
+
+type SentryEnvironment = 'development' | 'preview' | 'production';
+
+export class ErrorReporter {
+  private static readonly eventQueueTimeoutMs = 1000;
+
+  /**
+   * 'OnUncaughException' integration must be ignored since it causes Sentry to permanently fail during runtime. Default Sentry behaviour is to call process.exit() for uncaught errors. This is being prevented by VS Code and is logged as:
```
Comment (bastiandoetsch) [inline src/snyk/common/error/errorReporter.ts:16]: Please break the line :)

File: src/test/unit/common/error/integrations/onUncaughtException.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    const res = uncaughtExceptionIntegration.isExtensionOriginError(error);
+
+    strictEqual(res, true);
+  });
+
+  test('Is not extension origin error - another extension stacktrace', () => {
+    const error = new Error();
+    error.stack = `Error: test error
+    at Test.Runnable.run (/Users/snyk/.vscode/extensions/microsoft.sample-extension-1.2.4/out/extension.js:100:5)`;
```
Comment (bastiandoetsch) [inline src/test/unit/common/error/integrations/onUncaughtException.test.ts:26]: Maybe also test with Windows/Linux paths?

**Review by bastiandoetsch — APPROVED** [review]

File: src/test/unit/common/error/integrations/onUncaughtException.test.ts (hunk 3 above)
Comment (michelkaporin) [reply src/test/unit/common/error/integrations/onUncaughtException.test.ts:26]: fair point, did want to cover just the basics from now. I've added other paths now (Linux and MacOS have both the same extension folder path).

File: src/snyk/base/modules/interfaces.ts (hunk 1 above)
Comment (michelkaporin) [reply src/snyk/base/modules/interfaces.ts]: Good catch

File: src/snyk/common/error/errorReporter.ts (hunk 2 above)
Comment (michelkaporin) [reply src/snyk/common/error/errorReporter.ts]: made a break after "errors" as agreed in DM :) ✔️

**Pull request merged**